[Apps] [revampdoc] [publishyourapp] (#2821)
MicrosoftDocs/msteams-docs: msteams-platform/concepts/build-and-test/apps-package.md

# Create an app package for your Microsoft Teams app

This document guides you on how to create your app package.

> Apps in Teams are defined by an app manifest JSON file and bundled in an app package with their icons. You need an app package to upload and install your app in Teams and publish to an App Store or to AppSource.

Reviewer: ```suggestion
Apps in Teams are defined by an app manifest JSON file and bundled in an app package with their icons. You need an app package to upload and install your app in Teams and publish to the Teams store and AppSource.
```

A Teams app package is a .zip file containing the following:

* A manifest file named **manifest.json**, which specifies attributes of your app and points to needed resources for your experience. For example, location of the tab configuration page or the Microsoft app ID for the bot.
* [Color and outline icons for your app](#app-icons).

> ## Create a manifest

Reviewer: ```suggestion
## App manifest
```
At this point, your manifest has been created. I think maybe we only need to briefly describe what the manifest is (your app configurations) and how it's part of your package.

> Teams App Studio help you to configure your manifest. It also contains a react control library and configurable samples for cards. See [App Studio Overview](~/concepts/build-and-test/app-studio-overview.md).

Reviewer: I do not think we need to cover what App Studio is since the user likely will have used it to get to this point.

Your manifest file must be at the top level of the uploaded package with the name **manifest.json**. For Teams App Store and AppSource submission, use the current [manifest schema](~/resources/schema/manifest-schema.md).

> [!TIP]
> Mention the schema at the beginning of your manifest to enable IntelliSense or similar support from your code editor:
>

> > `"$schema": "https://developer.microsoft.com/json-schemas/teams/v1.8/MicrosoftTeams.schema.json",`

Reviewer: Remove.

## App icons

> Your app package must include two PNG versions of app icon. A color icon and an outline icon. 

Reviewer: ```suggestion
Your app package must include two PNG versions of your app icon: A color and outline version. 
```

> [!Note]
> If your app has a bot or messaging extension, the icons must be included in your [Microsoft Azure Bot Service registration](https://aka.ms/aadapplist).

> For your app to pass the AppSource review, these icons must meet the following size requirements:

Reviewer: ```suggestion
For your app to pass Teams store review, these icons must meet the following size requirements:
```

### Color icon

The color version of your icon displays in most Teams scenarios and must be 192x192 pixels. The icon symbol must be 96x96 pixels and can be any color. However, it must sit on a solid or fully transparent square background.

Teams automatically crops your icon to display a square with rounded corners in multiple scenarios and a hexagonal shape in bot scenarios. To crop the symbol without losing any detail, include 48 pixels of padding around your symbol.

:::image type="content" source="../../assets/images/icons/design-color-icon.png" alt-text="Teams color icon design guidance." border="false":::

### Outline icon

An outline icon displays in two scenarios:

* When your app is in use and hoisted on the left side of Teams app bar.
* When a user pins your app's messaging extension.

The icon must be 32x32 pixels. It can be white with a transparent background or transparent with a white background. No other colors are permitted. The outline icon must not have any extra padding around the symbol.

:::image type="content" source="../../assets/images/icons/design-outline-icon.png" alt-text="Teams color icon design guidance." border="false":::

> ## Best practices

Reviewer: ```suggestion
### Best practices
```
Changing back to an H3 since these pertain to the H2 "App icons".
